src/runtime/runtime_wasip1.go

```diff
@@ -100,6 +134,51 @@ func ticks() timeUnit {
 	return timeUnit(nano)
 }
 
+// Track which state we're in: before (or during) init, running inside
```

Comment:
Can these lines be moved out to a `runtime_wasm.go` file, so they can be shared with `wasip1`, `wasip2`, and `unknown`?

Reply:
Done, moved to runtime_wasmentry.go.